addons/hr_timesheet/project_timesheet.py

```diff
@@ -59,3 +59,23 @@ def _hours_get(self):
     subtask_count = fields.Integer(compute='_get_subtask_count', type='integer', string="Sub-task count")
 
     _constraints = [(osv.osv._check_recursion, 'Circular references are not permitted between tasks and sub-tasks', ['parent_id'])]
+
+class account_analytic_line(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.multi
+    def unlink(self):
+        for line in self:
+            if self.env.context.get("from_post"):
+                break
+            so_line = line.so_line
+            if not so_line and not so_line.state == 'sale' and not so_line.product_id.track_service in ['task']:
```

Comment:
What the what now? Why not just `if so_line.state != 'sale' and so_line.product_id.track_server != 'task'`?